EIPS/eip-7484.md

```diff
@@ -0,0 +1,239 @@
+
+A proposed solution to these security considerations is a permission framework that allows module developers to define the permissions that their modules require. The counterpiece of these permissions are attestations that assert statements about the security of modules and can be queried onchain.
+
+This proposal is independent of the exact implementation of the permissioning system and could even be used across different kinds of permissioning systems. Instead, the goal of this proposal is to outline a standard, but flexible, way to create onchain attestations and standardise how these attestations are queried by the smart account during module installation or execution.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Definitions
+
+- **Smart account** - An ERC-4337 compliant smart contract account that has a modular architecture.
+- **Module** - Self-contained smart account functionality.
+- **Attestation** - Onchain assertions made about the security of a module.
+- **Registry** - An onchain list of attestations about modules.
+- **Adapter** - Smart account functionality that handles the fetching and validation of attestations from the registry.
+- **Attester** - An entity that makes an attestation about a module.
+
+### Overview
+
+![Adapter flow](../assets/eip-7484/flow.png)
+
+An Adapter is a piece of functionality that allows smart accounts to query the module registry before using a module. In order to be compatible with this proposal, a smart account MUST implement the adapter either natively in the account or as a default module that is installed during account deployment.
+
+The adapter:
+
+- MUST query the registry at least once before calling a module for the first time.
+- MUST use one of the functions on the IRegistry interface when querying the registry.
+- Is RECOMMENDED to query the registry before module installation.
+- MAY query the registry before module execution.
+
+### Interfaces
+
+#### IRegistry.sol
+
+##### `check`
+
+Checks the attestation of a given module address and attestations made by a single attester.
+
+Returns two timestamps. `listedAt` is a `uint48` timestamp of when the attestation was made.
+The second return parameter `revokedAt` is a `uint48` timestamp of when an attestation was revoked.
+
+**NOTE**:
+
+- `listedAt` of 0 MUST be treated as a security risk.
+- `revokedAt` of non-0 MUST be treated as a security risk.
+
+```solidity
+function check(
+        address module,
+        address attester
```

Comment:
Will be there a public list to take attesters to check from?

Also, what if Smart Account does not care about which attester provided attestation for a module but only cares certain cumulative threshold of 'authority points' is reached?
Is it meant to be handled by an Adapter internally? Like maintaining such a list of attesters with their authority level and allowing enabling (executing) modules only after some threshold?